### PR TITLE
transfer the apikey with ciphertext to the enroll app

### PR DESCRIPTION
**Pull request by bajo14**
- process apikey encryption/decryption
- add sample_rijndael128GCM_decrypt

test: passed the uinttest

Signed-off-by: bajo14 <z_sr@msn.com>

### CODE REVIEW

File: core/App/ehsm_napi.cpp [hunk 1]
```diff
@@ -1326,6 +1326,7 @@ char *NAPI_RA_GET_API_KEY(const char *p_att_result_msg)
 
     char p_appid[UUID_STR_LEN] = {0};
     ehsm_data_t p_apikey;
+    ehsm_data_t cipherkey;
```
[the remark this answers is not in the document]
Comment (syan10) [reply core/App/ehsm_napi.cpp:1329]: cipherkey ->cipherapikey

File: core/App/ehsm_napi.cpp [hunk 2]
```diff
@@ -1326,6 +1326,7 @@ char *NAPI_RA_GET_API_KEY(const char *p_att_result_msg)
 
     char p_appid[UUID_STR_LEN] = {0};
```
Comment (syan10) [inline core/App/ehsm_napi.cpp:1327]: change the commit title as
"
transfer the apikey with ciphertext to the enroll app
"

File: core/App/ehsm_provider.cpp [hunk 3]
```diff
@@ -815,26 +815,30 @@ ehsm_status_t ExportDataKey(ehsm_keyblob_t *cmk,
         return EH_OK;
 }
 
-ehsm_status_t generate_apikey(ehsm_data_t *apikey)
+ehsm_status_t generate_apikey(ehsm_data_t *apikey, ehsm_data_t *cipherkey)
```
Comment (syan10) [inline core/App/ehsm_provider.cpp:818]: name not changed

File: core/Enclave/enclave_hsm.cpp [hunk 4]
```diff
@@ -789,16 +789,20 @@ sgx_status_t enclave_rsa_decrypt(const uint8_t *cmk_blob, size_t cmk_blob_size,
     return ret;
 }
 
-sgx_status_t enclave_generate_apikey(uint8_t *p_apikey, uint32_t apikey_len)
+sgx_status_t enclave_generate_apikey(sgx_ra_context_t context,
+                                     uint8_t *p_apikey, uint32_t apikey_len,
+                                     uint8_t *cipherkey, uint32_t cipherkey_len)
```
Comment (syan10) [inline core/Enclave/enclave_hsm.cpp:794]: same here

File: ehsm_kms_service/function.js [hunk 5]
```diff
@@ -151,7 +151,7 @@ function store_cmk(napi_res, res, appid, payload, DB) {
       alias: '',
       keyspec,
       origin,
-      keyState: 1,
+      keyState: true,
```
[the remark this answers is not in the document]
Comment (wanghouqi) [reply ehsm_kms_service/function.js:154]: typo.It will be restore.

**Comment by bajo14** [comment]
commit new pr

**Pull request closed without merging**